application/src/main/java/com/speedment/jpastreamer/application/JPAStreamer.java

```diff
@@ -190,6 +196,91 @@ default <T> Stream<T> stream(final Projection<T> projection) {
         return stream(StreamConfiguration.of(projection.entityClass()).selecting(projection));
     }
 
+    /**
+     * Creates and returns a new {@link StreamSupplier} that can create 
+     * {@link Stream}s over all entities in the underlying data source (e.g database) 
+     * of the provided type {@code entityClass}. 
+     * <p>
+     * The provided {@link StreamSupplier} will <em>not</em> be closed whenever
+     * the generated {@link Stream} instance is closed.
+     * <p>
+     * If you are using the same Stream source frequently e.g. Film.class, 
+     * consider configuring a {@link StreamSupplier} that can supply 
+     * {@link Stream}s from the same source over and over again. 
+     * This save resources and avoids instantiating a new {@link EntityManager} for each new {@link Stream}s.
+     * <p>
+     * Here is an example of using a {@link StreamSupplier}: 
+     * <pre>{@code
+     *    final StreamSupplier<Film> streamSupplier = jpaStreamer.createStreamSupplier(Film.class);
```

Comment:
I think using Try-With-Resources would be better here.

Reply:
Could I provide both examples? I just think this way is clearer, but I totally get that TWR is a better practise.